services/graph/pkg/identity/ldap.go

```diff
@@ -802,3 +892,129 @@ func replaceDN(fullDN *ldap.DN, newDN string) (string, error) {
+func (i *LDAP) CreateLDAPGroupByDN(dn string) error {
+	ar := ldap.NewAddRequest(dn, nil)
+
+	attrs := map[string][]string{
+		"objectClass": {"groupOfNames", "top"},
+		"member":      {""},
+	}
+
+	for attrType, values := range attrs {
+		ar.Attribute(attrType, values)
+	}
+
+	return i.conn.Add(ar)
+}
+
+func (i *LDAP) disableUser(logger log.Logger, userDN string) (err error) {
+	group, err := i.getEntryByDN(i.localUserDisableGroupDN, []string{_ldapGroupMemberAttribute}, _ldapGroupOfNamesAttribute)
+
+	if err != nil {
+		return err
+	}
+
+	mr := ldap.ModifyRequest{DN: group.DN}
+	mr.Add(_ldapGroupMemberAttribute, []string{userDN})
+
+	err = i.conn.Modify(&mr)
+	var lerr *ldap.Error
+	if errors.As(err, &lerr) {
+		// If the user is already in the group, just log a message and return
+		if lerr.ResultCode == ldap.LDAPResultAttributeOrValueExists {
+			logger.Info().Msg("User already in group for disabled users")
+			return nil
+		}
+	}
+
+	return err
+}
+
+func (i *LDAP) enableUser(logger log.Logger, userDN string) (err error) {
+	group, err := i.getEntryByDN(i.localUserDisableGroupDN, []string{_ldapGroupMemberAttribute}, _ldapGroupOfNamesAttribute)
+
+	if err != nil {
+		return err
+	}
+
+	mr := ldap.ModifyRequest{DN: group.DN}
+	mr.Delete(_ldapGroupMemberAttribute, []string{userDN})
+
+	err = i.conn.Modify(&mr)
+	var lerr *ldap.Error
+	if errors.As(err, &lerr) {
+		// If the user is not in the group, just log a message and return
+		if lerr.ResultCode == ldap.LDAPResultNoSuchAttribute {
+			logger.Info().Msg("User was not in group for disabled users")
+			return nil
+		}
+	}
+
+	return err
+}
+
+func (i *LDAP) userEnabledByAttribute(user *ldap.Entry) bool {
+	enabledAttribute := booleanOrNil(user.GetEqualFoldAttributeValue(i.userAttributeMap.accountEnabled))
+
+	if enabledAttribute == nil {
+		return true
+	}
+
+	return *enabledAttribute
+}
+
+func (i *LDAP) usersEnabledStateFromGroup(users []string) (usersEnabledState map[string]bool, err error) {
+	group, err := i.getEntryByDN(i.localUserDisableGroupDN, []string{_ldapGroupMemberAttribute}, _ldapGroupOfNamesAttribute)
+
+	if err != nil {
+		return nil, err
+	}
+
+	usersEnabledState = make(map[string]bool, len(users))
+	for _, user := range users {
+		usersEnabledState[user] = true
+	}
+
+	for _, memberDN := range group.GetEqualFoldAttributeValues(_ldapGroupMemberAttribute) {
+		usersEnabledState[memberDN] = false
+	}
+
+	return usersEnabledState, err
+}
+
+// UserEnabled returns if a user is enabled. This can depend on a flag in the user entry or group membership
+func (i *LDAP) UserEnabled(user *ldap.Entry) (bool, error) {
+	usersEnabledState, err := i.usersEnabledState([]*ldap.Entry{user})
+
+	if err != nil {
+		return false, err
+	}
+
+	return usersEnabledState[user.DN], nil
+}
+
+func (i *LDAP) usersEnabledState(users []*ldap.Entry) (usersEnabledState map[string]bool, err error) {
+	usersEnabledState = make(map[string]bool, len(users))
```

Comment:
In order to have the behavior consistent with the related reva change. We should not evaluate the enabled state from the user attributes when using `DisableMechanismGroup`.

I'd suggest handling this in a switch:

```
switch i.disableUserMechanism {
case DisableMechanismGroup:
    // handle group stuff here
case DisableMechanismAttribute:
    // handle userattribute here
}
```